plugins/http_plugin/include/eosio/http_plugin/http_plugin.hpp

```diff
@@ -166,23 +186,25 @@ namespace eosio {
    };
 
    template<typename T, http_params_types params_type>
-   T parse_params(const std::string& body) {
+   T parse_params(std::string& body) {  // needs to trim space,
```

Comment:
While I don't think this will have issues, I don't like the practice of mutating inputs in a way that is not clear to the caller.  

Please restore the `const` qualifier on the input and either change to `boost::algorithm::trim_copy` or create a `std::string_view` of the equivalent sub-string.  The former is fine unless we are really concerned with the extra performance impact of the copy. 

Unfortunately, I don't know of a clean algorithm to do so and neither `std::string_view` or `boost::string_view` is compatible with `boost::algorithm::trim(_copy)`

Reply:
I don't see a clean way to trim w/o copy or modifying, or implement a string_view trim.

Reply:
Using a trimmed string_view, so no change to user input data